app/controllers/service_calendars_controller.rb

```diff
@@ -139,60 +106,72 @@ def move_visit_position
   end
 
   def toggle_calendar_row
-    @line_items_visit  = LineItemsVisit.find(params[:line_items_visit_id])
-    @service           = @line_items_visit.line_item.service if params[:check]
-    @portal            = params[:portal] == 'true'
-
-    return unless @line_items_visit.sub_service_request.can_be_edited? || @portal
-
-    @line_items_visit.visits.each do |visit|
-      if params[:check]
-        visit.update_attributes(quantity: @service.displayed_pricing_map.unit_minimum, research_billing_qty: @service.displayed_pricing_map.unit_minimum, insurance_billing_qty: 0, effort_billing_qty: 0)
-      elsif params[:uncheck]
-        visit.update_attributes(quantity: 0, research_billing_qty: 0, insurance_billing_qty: 0, effort_billing_qty: 0)
-      end
+    @admin              = params[:admin] == 'true'
+    @tab                = 'template'
+    @page               = params[:page]
+    @line_items_visit   = LineItemsVisit.eager_load(sub_service_request: { organization: { parent: { parent: :parent } } }, line_item: [:admin_rates, service: [:pricing_maps, organization: [:pricing_setups, parent: [:pricing_setups, parent: [:pricing_setups, parent: :pricing_setups]]]]]).find(params[:line_items_visit_id])
+    @arm                = @line_items_visit.arm
+    @line_items_visits  = @arm.line_items_visits.eager_load(line_item: [:admin_rates, service: [:pricing_maps, organization: [:pricing_setups, parent: [:pricing_setups, parent: [:pricing_setups, parent: :pricing_setups]]]], service_request: :protocol])
+    @visit_groups       = @arm.visit_groups.paginate(page: @page.to_i, per_page: VisitGroup.per_page).eager_load(visits: { line_items_visit: { line_item: [:admin_rates, service: [:pricing_maps, organization: [:pricing_setups, parent: [:pricing_setups, parent: [:pricing_setups, parent: :pricing_setups]]]], service_request: :protocol] } })
+    @visits             = @line_items_visit.visits.eager_load(service: :pricing_maps)
+    @locked             = !@admin && !@line_items_visit.sub_service_request.can_be_edited?
+
+    if params[:check] && !@locked
+      unit_minimum = @line_items_visit.line_item.service.displayed_pricing_map.unit_minimum
+
+      @visits.update_all(quantity: unit_minimum, research_billing_qty: unit_minimum, insurance_billing_qty: 0, effort_billing_qty: 0)
+    elsif params[:uncheck] && !@locked
+      @visits.update_all(quantity: 0, research_billing_qty: 0, insurance_billing_qty: 0, effort_billing_qty: 0)
     end
 
     # Update the sub service request only if we are not in dashboard; admin's actions should not affect the status
```

Comment:
This logic was incorrect. Should not update only if you are an _admin_ or the SSR is _locked_